### PR TITLE
[BENCH-897] Add Data Storage Volume to GCP Controlled Resource

### PR DESCRIPTION
**Pull request by iliadsh**

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java [hunk 1]
```diff
@@ -193,7 +194,15 @@ static Instance setFields(
...
                         .setSourceImage(creationParameters.getVmImage())
-                        .setDiskSizeGb(Long.valueOf(100)))));
+                        .setDiskSizeGb(Long.valueOf(100))),
+            new AttachedDisk()
+                .setBoot(false)
+                .setDeviceName(DATA_DISK_NAME)
+                .setAutoDelete(true)
+                .setInitializeParams(
+                    new AttachedDiskInitializeParams()
+                        .setDiskType(creationParameters.getDataDiskType())
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:204]: should this be conditional? 

only add an AttachedDisk if the creationParameters contains datadisktype and datadisksizegb

**Review by yuhuyoyo — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java (hunk 1 above)
Comment (iliadsh) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:204]: My impression from John's doc was that all instances would have an extra PD... @jmczerk confirm?

Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:204]: or a default type and size?

Comment (iliadsh) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java:204]: Hm, but GCP will default their values if none are provided.

**Pull request merged**